passenger-service/src/main/java/com/modsen/passengerservice/services/PassengerService.java

```diff
@@ -0,0 +1,242 @@
+        return HttpStatus.OK;
+    }
+
+    public HttpStatus deletePassengerById(Long passengerId) throws PassengerNotFoundException{
+        return delete(passengerId,String.format("passenger with id: %s is not found.",
+                passengerId),passengerRepository::findById);
+    }
+
+    public HttpStatus deletePassengerByPhone(String phone) throws PassengerNotFoundException{
+        return delete(phone,String.format("passenger with phone: %s is not found.",
+                phone),passengerRepository::findByPhone);
+    }
+
+    public HttpStatus deletePassengerByEmail(String email) throws PassengerNotFoundException{
+        return delete(email,String.format("passenger with email: %s is bot found.",
+                email),passengerRepository::findByEmail);
+    }
+
+    public HttpStatus deletePassengerByUsername(String username) throws PassengerNotFoundException{
+        return delete(username,String.format("passenger with username: %s is not found.",
+                username),passengerRepository::findByUsername);
+    }
+
+    public HttpStatus updatePassengerByEmail(PassengerDto passengerDto)
+            throws PassengerNotFoundException{
+        return update(passengerRepository::findByEmail,
+                passengerDto::getEmail,
+                passengerDto,
+                String.format("passenger with email: %s is not found.", passengerDto.getEmail()));
+    }
+
+    public HttpStatus updatePassengerByPhone(PassengerDto passengerDto)
+            throws PassengerNotFoundException{
+        return update(
+                passengerRepository::findByPhone,
+                passengerDto::getPhone,
+                passengerDto,
+                String.format("passenger with phone: %s is not found.", passengerDto.getPhone())
+        );
+
+    }
+
+    public HttpStatus updatePassengerByUsername(PassengerDto passengerDto)
+            throws PassengerNotFoundException{
+        return update(
+                passengerRepository::findByUsername,
+                passengerDto::getUsername,
+                passengerDto,
+                String.format("passenger with username: %s is not found.", passengerDto.getUsername())
+        );
+
+    }
+
+    public ResponseEntity<Page<PassengerDto>> getLimitedList(int offset, int limit){
+        Pageable pageable = PageRequest.of(offset,limit);
+        Page<Passenger> passengerPage = passengerRepository.findAll(pageable);
+        Page<PassengerDto> result = passengerPage.map(passengerMapper::entityToDto);
+        return ResponseEntity.ok(result);
+    }
+
+    private HttpStatus update(Function<String, Optional<Passenger>> repositoryFunc,
+                                      Supplier<String> passengerDtoGetter,
+                                      PassengerDto passengerDto,
+                                      String exceptionMessage) throws PassengerNotFoundException{
+        Optional<Passenger> passenger_opt = repositoryFunc.apply(passengerDtoGetter.get());
```

Comment:
Если сделать так, то тот некрасивый if перестанет быть нужным
```suggestion
        Passenger passenger_opt = repositoryFunc.apply(passengerDtoGetter.get())
                                                .orElseThrow( () -> new PassengerNotFoundException(exceptionMessage));
```